m3/reporter.go

```diff
@@ -288,45 +323,57 @@ func (r *reporter) AllocateHistogram(
 	buckets tally.Buckets,
 ) tally.CachedHistogram {
```

Comment:
Note: as a followup, I'll probably refactor histograms to share the same tag storage and store their separate bucket names (instead of separate tag maps), and resolve the tags at reporting time (potentially via map pooling) instead of allocating like crazy.